File: Swift/two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict = [Int: Int]()
        
        for (i, num) in nums.enumerated() {
            if let index = dict[target - num] {
                return [index, i]
            }
            dict[num] = i
        }
        
        fatalError("no valid outputs")
    }
}

File: Python/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        d = dict()

        for i in range(len(nums)):
            if (target - nums[i]) in d:
                return [d[target - nums[i]], i]
            d[nums[i]] = i
